ampd/src/commands/bond_worker.rs

```diff
@@ -0,0 +1,48 @@
+use std::path::Path;
+
+use cosmrs::cosmwasm::MsgExecuteContract;
+use cosmrs::tx::Msg;
+use cosmrs::Coin;
+use error_stack::Result;
+use report::ResultCompatExt;
+use service_registry::msg::ExecuteMsg;
+use valuable::Valuable;
+
+use crate::commands::{broadcast_tx, worker_pub_key};
+use crate::config::Config;
+use crate::{Error, PREFIX};
+
+#[derive(clap::Args, Debug, Valuable)]
+pub struct Args {
+    pub amount: u128,
+    pub denom: String,
+}
+
+pub async fn run(config: Config, state_path: &Path, args: Args) -> Result<(), Error> {
+    let coin = Coin::new(args.amount, args.denom.as_str()).change_context(Error::Command)?;
+
+    let pub_key = worker_pub_key(state_path, config.tofnd_config.clone()).await?;
+
+    let msg = serde_json::to_vec(&ExecuteMsg::BondWorker {
+        service_name: config.service_registry.service_name.clone(),
+    })
+    .expect("bond worker msg should serialize");
+
+    let tx = MsgExecuteContract {
+        sender: pub_key
+            .account_id(PREFIX)
+            .expect("failed to convert to account identifier"),
```

Comment:
I commented on this in the previous command PR already. I don't like this implicit coupling, relying on tofnd to provide the correct key. Two options: either return an error instead of panicking, or create type wrappers so it's guaranteed that account_id will be called with the correct key type

Reply:
will error out for now, prefer to create a type wrapper as we know tofn client returns an ecdsa key, error if it fails to parse